### PR TITLE
Fix 8162: project settings layout issues

### PR DESCRIPTION
**Pull request by benjaminapetersen**
Fixes issue #8162

Primarily addresses stacking the form for mobile. I haven't been able to reproduce the overlapping text via various mobile browsers on browserstack.  Not sure yet what the deal is there.

@spadgett @jwforres @sg00dwin

### CODE REVIEW

**Comment by spadgett** [comment]
Still seeing the issue :(

It appears to be caused by the `grow="4"` attribute.

**Comment by benjaminapetersen** [comment]
@spadgett yup, got ahold of it. I've been able to reproduce it inconsistenly across simulator + browserstack iOS devices.  

@sg00dwin @rhamilto bringing you guys into this one.  I made a new less file called `_spacing.less` and created a couple recursive mixins to generate generic spacer classes for us, since we only have a couple right now (`.gutter-top .gutter-bottom .gutter-top-bottom`).  They are similar to what may be coming down the line via [Boostrap 4](http://v4-alpha.getbootstrap.com/components/utilities/#spacing) (is alpha now).

Example:
```html
<div row flex class="mar-md mar-top-none">Some text.</div>
```

**Comment by rhamilto** [comment]
I wonder if we should we adopt the class names that Bootstrap 4 is using?  That way, when we switch to Bootstrap 4, there is no need to replace our addition with Bootstrap's?

**Comment by benjaminapetersen** [comment]
@rhamilton, i considered that.  I'll throw a few thoughts out to chew on, but if we are cool with these I'm good with rolling with everyone's preference.

The classnames I used are in line with a branch in the [layout.attrs](https://github.com/benjaminapetersen/layout.attrs) repo that hasn't been merged in yet.  Its something I've been tinkering with for a while.

From the bootstrap version (`.m-l-1`, `.p-x-2`, etc) I do like how terse the classnames are.  I'm not in love with single letters, though, and I definitely don't like the 0,1,2,3, etc as size markers as it makes me think of px values.  

I adopted the `sm, md, lg, xl`, etc because that is familiar terminology & it keeps specific px values out of markup.  If we used variables we could potentially alter these (like how we can customize boostrap columns and other things).

Side by side:

```css
// bootstrap
.m-t-0
.p-b-1

// mine
.mar-sm
.pad-md
.pad-top-lg

```
Btw, bootstrap seems to only have these utilities for individual sides, I don't think you can `class="pad-md pad-top-lg` override.

**Comment by spadgett** [comment]
Fields no longer line up in edit mode :/  I feel like we should simply use a `dl` so it uses consistent styles with the browse pages.

openshift_web_console

+1 to the padding utilities, although depending on the fix we settle on here, it might be better as a separate pull request.

**Comment by benjaminapetersen** [comment]
Ah, they line up on mobile, my last change broke non-mobile, sorry will fix, just need a `mar-right-none` on 'Display Name'.  

I moved away from the `dl, dd, dt` to have the flexibility of additional markup.  I remember trying to work with the original dl but having issues with that as well toggling back and forth between edit modes.  I can try to put it back if you think thats ultimately better.

Yeah, wanted to have some discussion around those utils to see what we should do (separate PR, rename, etc)

File: assets/app/styles/_spacing.less [hunk 1]
```diff
@@ -0,0 +1,36 @@
+// Spacing utility classes
+// creates:
+//  .pad-sm, .pad-md, .pad-lg, etc
+//  .mar-sm, .mar-md, .mar-lg, etc
+//  .pad-left-sm, .mar-left-sm, etc
+//
+@sizes: none 0px e('!important'), sm 4px e(''), md 8px e(''), lg 12px e(''), xl 16px e(''), xxl 32px e('');
```
Comment (rhamilto) [inline assets/app/styles/_spacing.less]: Sorry.  This is a pet peeve.  Please drop "px" from 0px.  0 is always 0.  :)

Comment (liggitt) [inline assets/app/styles/_spacing.less]: Unless it's temperature (though that's not until CSS4)

Comment (sg00dwin) [inline assets/app/styles/_spacing.less]: max-high: 120°;
transform: rotate(0.5turn);
display: simmer;

Comment (benjaminapetersen) [inline assets/app/styles/_spacing.less]: ok :smile:
Comments for that can prob move [to the PR against console_3.1 branch](https://github.com/jwforres/origin/pull/26), I nixed these helper classes here so we can debate about `px` and other evils for as long as we want.

**Comment by benjaminapetersen** [comment]
Nixed the margin/padding utils, have a temp less file with classes i'd replace later when we decide on the utils.

Some screenshots for where this is at:

screen shot 2016-03-24 at 4 20 51 pm

screen shot 2016-03-24 at 4 20 56 pm

Mobile:

screen shot 2016-03-24 at 4 21 09 pm

screen shot 2016-03-24 at 4 21 02 pm

iOS Simulator:

screen shot 2016-03-24 at 4 26 16 pm

screen shot 2016-03-24 at 4 26 27 pm

Thoughts? There is some slight difference between mobile/simulator. Everything still lines up, but a little more white space.

File: assets/app/styles/_settings.less [hunk 2]
```diff
@@ -0,0 +1,13 @@
+// NOTE: these are temporary classes
+// will to replace with reusable generic utils
+.project-item {
+  margin:4px 8px;
+}
+
+.project-item-left-none {
+  margin-left: 0px !important;
```
Comment (spadgett) [inline assets/app/styles/_settings.less]: @rhamilto would disapprove

Comment (benjaminapetersen) [inline assets/app/styles/_settings.less]: Yikes!  I will fix.

**Comment by benjaminapetersen** [comment]
Latest tweaks look like:

screen shot 2016-03-28 at 3 05 07 pm

screen shot 2016-03-28 at 3 05 15 pm

**Comment by spadgett** [comment]
[merge]

**Comment by spadgett** [comment]
@benjaminapetersen bindata conflict

**Comment by benjaminapetersen** [comment]
Ah, updated.

**Comment by spadgett** [comment]
[merge]

**Pull request merged**